### PR TITLE
chore: update CD to release beta version from prerelease branch

### PR DESCRIPTION
**Pull request by wenytang-ms**
currently the following packages update the version to latest alpha.
cli 2.0.0-alpha
fx-core 2.0.0-alpha
vscode-extension 4.99.0-alpha
server 2.0.0-alpha

from now on, prerelease branch only release production with preid as beta

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -89,9 +84,14 @@ jobs:
...
-        if: ${{ github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/prerelease' }}
+        if: ${{ github.ref == 'refs/heads/dev' }}
         run: |
-          npx lerna version prerelease --preid=alpha.$(git rev-parse --short HEAD) --exact --no-push --allow-branch ${GITHUB_REF#refs/*/} --yes
+          npx lerna version prerelease --preid=alpha.$(git rev-parse --short HEAD) --exact --no-push --allow-branch dev --yes
+
+      - name: release beta packages to npmjs.org
+        if: ${{ github.ref == 'refs/heads/prerelease' }}
+        run: |
+          npx lerna version prerelease --preid=beta.$(date "+%Y%m%d%H") --exact --no-push --allow-branch prerelease --yes
```
Comment (IvanJobs) [inline .github/workflows/cd.yml:94]: I'm afraid if two beta would be released in one hour so that the preid would have conflicts then.

**Review by IvanJobs — APPROVED** [review]
LGTM

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -238,9 +234,16 @@ jobs:
           git commit -m "chore: update ai key"
 
       - name: publish alpha release to npm org
-        if: ${{ github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/prerelease' }}
+        if: ${{ github.ref == 'refs/heads/dev'}}
+        run: |
+          npx lerna publish from-package --dist-tag=alpha --yes --allow-branch dev
+        env:
+          TEMPLATE_PATH: ${{ runner.temp }}/teamsfx_templates_v3;${{ runner.temp }}/teamsfx_templates
```
Comment (hund030) [inline .github/workflows/cd.yml:241]: Remove teamsfx_templates folder?

Comment (wenytang-ms) [reply .github/workflows/cd.yml:272]: removed

**Review by hund030 — APPROVED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (wenytang-ms) [reply .github/workflows/cd.yml:94]: even release twice in one hour, still success on pipeline

**Review by IvanJobs — APPROVED** [review]

**Pull request merged**